report/report.tex

```diff
@@ -348,6 +361,14 @@ \section{Reflection}
         Although everybody was aware of the amount of work that had to be put into the Eval module from the get go, it still cost a lot more time eventually. For a problem that, we all already recognised back then, was going to be difficult, more time should have been claimed since a problem that seems theoretically challenging is almost always far more difficult in practice. However, a mistake was also made by not correctly working in a strictly iterative fashion. The result of this was that the Eval module was almost always broken in one way or another because it was still not finished. Working in a strictly iterative fashion, always working on one function and one function only, would have resulted in a clearer overview of what the tasks and goals were at every moment, and in clearer moments on which the only partially finished, but stable, Eval module could be merged with the master branch.
         
 Due to an initial lack of tests for all possible Hanus constructs, progress on the evaluator was slightly delayed towards the end of the project, since a few previously unknown bugs in the parser became exposed at that time. 
+\section{Future work}
+	The Hanus project can be extended or improved in multiple ways. Some ideas for future work on Hanus are:
+        \begin{enumerate}
+                	\item Implementing proper field and array indexers, i.e. at the moment, multiple concatenated field and array indexers are not supported.
+                	\item Better error messages that don't look like Haskell errors. This is not a bug per se, but when a programmer writes complicated Hanus code, they will quickly find out that the error messages are very difficult to debug, partially because the error messages concern the generated Haskell code, but more so because of the fact that the structure of certain Hanus expressions such as loops is rigorously changed in the compilation step to Haskell, making it very hard for the programmer to find out what is wrong in the actual Hanus program.
+                	\item Template Haskell bug fixes. During the development of Hanus, multiple Template Haskell bugs have been found that are most probably caused by template Haskell, e.g. function declarations that Haskell cannot find when they are declared at the top level, but that Haskell \emph{can} find when they are defined in a where clause of the callee, or variables that should get a unique name from Template Haskell, but still cause ``multiple declarations of x" errors in Haskell.
+                	\item Finding and fixing parser loop bugs. Although the parser has been tested rigorously, there are rare cases where the parser gets stuck in an endless loop. These bugs are very hard to find, so the chance that they were \emph{all} caught thanks to the many parser tests is still slim.
```

Comment:
I think that the parser will currently only have trouble with invalid strings, and that the error correction causes the loops.